cache_test.go

```diff
@@ -1086,3 +1088,66 @@ func TestUpdate(t *testing.T) {
 	found, replaced, err = cache.Update(key, updater)
 	assertExpectations(4, true, false, val2, val2)
 }
+
+func TestBackup(t *testing.T) {
+	iters := 100
+
+	cache := NewCache(1024)
+	for i := 0; i < iters; i++ {
+		key := mrand.Int()
+		val := strconv.Itoa(i)
+		err := cache.SetInt(int64(key), []byte(val), 60)
+		if err != nil {
+			t.Errorf("err: %s", err)
+		}
+	}
+
+	f, err := os.Create("/tmp/cache_backup.bin")
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	defer f.Close()
+
+	gz := gzip.NewWriter(f)
+	defer gz.Close()
+
+	err = cache.Backup(gz)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+}
+
+func TestRestore(t *testing.T) {
+	iters := 100
+	cache := NewCache(1024)
+
+	f, err := os.Open("/tmp/cache_backup.bin")
```

Comment:
This test case depends on the TestBackup has run first.
It would be better if we merge them into one test case TestBackupRestore.

And the ExpireAt related logic is not covered in the test.

Reply:
if it would be better together, can I cover `ExpireAt` in the test, assuming that the backup and restore do not take more than 1s

```go
totalExpire := 15

time.Sleep(time.Duration(minTimeExpire+totalExpire) * time.Second)

for i := 0; i < iters; i++ {
	key := mrand.Int()
	val := strconv.Itoa(i)
	val2, err := cache.GetInt(int64(key))

	if i < totalExpire {
		if err != nil && err != ErrNotFound {
			t.Errorf("err: %s", err)
		}
		continue
	}

	if err != nil {
		t.Errorf("err: %s", err)
	}

	if string(val2) != val {
		t.Errorf("err: %v %q==%q", key, val, val2)
	}

}
```